### PR TITLE
37594 sliceviewer sample data order

### LINKED ISSUE #37594: Slice Viewer indexOfValue - value out of range error
**Describe the bug**
There is an error when hovering over a specific dataset in the Slice Viewer. The error says:
```
Traceback (most recent call last):
  File "C:\Users\mlc47243\AppData\Local\mambaforge\envs\mantid-developer\lib\site-packages\matplotlib\cbook\__init__.py", line 314, in process
    func(*args, **kwargs)
  File "c:\users\mlc47243\documents\mantid-dev\mantid-review\mantid\qt\python\mantidqt\mantidqt\widgets\sliceviewer\views\cursor.py", line 66, in on_mouse_move
    self.on_cursor_at(xdata, ydata)
  File "c:\users\mlc47243\documents\mantid-dev\mantid-review\mantid\qt\python\mantidqt\mantidqt\widgets\sliceviewer\presenters\imageinfowidget.py", line 88, in _on_cursor_at_axesimage
    self._widget.cursorAt(xdata, ydata, arr[i, j], extra_cols)
Exception: indexOfValue - value out of range
```

**To Reproduce**
1. Load the below data into Workbench
[osiris100318-100329_graphite002_red_elwin_eq.zip](https://github.com/user-attachments/files/15928583/osiris100318-100329_graphite002_red_elwin_eq.zip)

3. Right click the workspace and `Open Slice Viewer`
4. Double click the y axis and change the limits to be from 262 to 266
5. Hover over the thin strip of data
6. The error will happen!

**Expected behavior**
There should be no error

**Screenshots**
image

**Platform/Version (please complete the following information):**
 - OS: Found on Windows

### PR DESCRIPTION
**Pull request by Despiix**
#### Summary of work
 Slice viewed displays error when sample data (Y axis) is unordered. Concluded to just displaying a warning when the unusual data is imported telling the user to order the relevant data themselves in necessary. 

Fixes #37594. 

 @robertapplin 

### To test:

1. **Import the attached data:**
File: [osiris100318-100329_graphite002_red_elwin_eq.zip](https://github.com/user-attachments/files/16931474/osiris100318-100329_graphite002_red_elwin_eq.zip)

2. **Open the Slice Viewer:**
Right click the data and click "Slice Viewer"

3. **Check for a Warning:**
The window should display the data, but also show a warning message on the side. 
Expected behavior: The data should appear "broken" (Thin line strip), and the warning should provide further details.

4. **Test with Other Data:**
Try importing and viewing other data files

---

### Reviewer

Please comment on the points listed below ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)).
**Your comments will be used as part of the gatekeeper process, so please comment clearly on what you have checked during your review.** If changes are made to the PR during the review process then your final comment will be the most important for gatekeepers. In this comment you should make it clear why any earlier review is still valid, or confirm that all requested changes have been addressed.

#### Code Review

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Do the release notes conform to the [release notes guide](https://developer.mantidproject.org/Standards/ReleaseNotesGuide.html)?

#### Functional Tests

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### Gatekeeper

If you need to request changes to a PR then please add a comment and set the review status to "Request changes". This will stop the PR from showing up in the list for other gatekeepers.

### CODE REVIEW

File: qt/applications/workbench/workbench/plugins/workspacewidget.py [hunk 1]
```diff
@@ -260,6 +260,15 @@ def _do_slice_viewer(self, names):
         parent, flags = get_window_config()
         for ws in self._ads.retrieveWorkspaces(names, unrollGroups=True):
             try:
+                specs = ws.getSpectrumNumbers()
+                y = ws.getYDimension()
+                if y.getName() == "sample":
+                    spec_to_y_dim = {s: y.getX(s) for s in specs}
```
Comment (RichardWaiteSTFC) [inline qt/applications/workbench/workbench/plugins/workspacewidget.py:266]: Not sure the benefit of using a `dict` here - the keys don't seem to be used.
I would probably do
```suggestion
                    spec_xvals = [ws.getYDimension().getX(ispec) for ispec in range(ws.getNumberHistograms())]
```
Note that `getX` requires the spectrum index not spectrum number.
Then you don't need to access `.values()` to check it's sorted, you could do
```
if sorted(spec_xvals) != spec_xvals:
```

**Review by RichardWaiteSTFC — CHANGES_REQUESTED** [review]
Warning shown when following instructions, as discussed offline this is a temporary measure, the bug is somewhere in the sliceviewer.
Just some comments about confusing index and spectrum number and a suggested change. I think this also requires a release note.

**Comment by sf1919** [comment]
@Despiix   - This is a victim of changes to pre-commit. See https://github.com/mantidproject/mantid/pull/37960#pullrequestreview-2298555084 for details of the changes that went in and see if you need to modify any of your files as a result and push a new commit.

**Review by RichardWaiteSTFC — DISMISSED** [review]
Looks good thanks for changes - maybe we can hunt this bug down during release

**Review by RichardWaiteSTFC — APPROVED** [review]
Approving again after target branch changed

**Pull request merged**